### PR TITLE
Removed support & specs for chefv12 and lower

### PR DESCRIPTION
**Pull request by dheerajd-msys**
Signed-off-by: dheerajd-msys <dheeraj.dubey@msystechnologies.com>

## Description
Removed support and specs for chef12 and lower versions.

## Issues Resolved
Fixes [chef/knife-windows#474](https://github.com/chef/knife-windows/issues/474)

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -125,7 +121,6 @@ def sample_data(file_name)
   config.filter_run_excluding :chef_gte_12_5_only => true unless chef_gte_12_5?
```
Comment (tas50) [inline spec/spec_helper.rb]: everything here but gte13 and the 14s can go away

Comment (dheerajd-msys) [reply spec/spec_helper.rb]: Thanks @tas50, Updated.

**Comment by dheerajd-msys** [comment]
@tas50, @lamont-granquist Could you take a look on this PR?

File: spec/functional/bootstrap_download_spec.rb [hunk 2]
```diff
@@ -184,11 +171,6 @@ def run_download_scenario
       @template_output = sample_data('win_template_rendered_with_bootstrap_install_command.txt')
     end
 
-    it "bootstrap_install_command option is rendered in the windows-chef-client-msi.erb template", :chef_lt_12_5_only => true do
-      expect(bootstrap.send(:render_template,@template_input)).to eq(
-        @template_output)
-    end
-
     context "when running chef-client 12.5.0 or greater", :chef_gte_12_5_only => true, :chef_lt_14_only => true  do
```
Comment (Vasu1105) [inline spec/functional/bootstrap_download_spec.rb]: Can we just verify this one or make changes to work from gte 13 and remove gte_12_5_only ?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -80,30 +80,10 @@ def windows2016?
...
-end
-
-def chef_gte_12?
-  Chef::VERSION.split('.').first.to_i >= 12
-end
-
-def chef_gte_12_5?
-  Gem::Version.new(Chef::VERSION) >= Gem::Version.new('12.5')
-end
-
```
Comment (Vasu1105) [inline spec/spec_helper.rb]: We removed this here so according to us this condition is not working in https://github.com/chef/knife-windows/pull/475/files#diff-1c44cb7c409c533aa88dc2f6ea3c1802R174.  I guess we should verify this.

File: spec/functional/bootstrap_download_spec.rb (hunk 2 above)
Comment (dheerajd-msys) [reply spec/functional/bootstrap_download_spec.rb]: Thank you @Vasu1105, Updated and removed extra template file and filter used here.

**Pull request merged**